wemake_python_styleguide/transformations/ast/enhancements.py

```diff
@@ -96,3 +138,38 @@ def _apply_if_statement(statement: ast.If) -> None:
             if child in statement.orelse:
                 setattr(statement, 'wps_if_chained', True)  # noqa: B010
                 setattr(child, 'wps_if_chain', statement)  # noqa: B010
+
+
+def _evaluate_node(node: ast.AST) -> Optional[Union[int, float, str, bytes]]:
+    """Returns the value of a node or its evaluation."""
+    if isinstance(node, ast.BinOp):
+        return _evaluate_operation(node)
+    if isinstance(node, (ast.Str, ast.Bytes)):
+        return node.s
+    if isinstance(node, ast.Name):
+        return None
+    try:
+        signed_node = literal_eval_with_names(node)
+    except Exception:
+        return None
+    return signed_node
+
+
+def _evaluate_operation(
+    statement: ast.BinOp,
+) -> Optional[Union[int, float, str, bytes]]:
+    """Tries to evaluate all math operations inside the statement."""
+    left = _evaluate_node(statement.left)
+    right = _evaluate_node(statement.right)
+    op = _AST_OPS_TO_OPERATORS.get(type(statement.op))
+
+    if left and right and op:
+        try:
+            evaluation = op(left, right)
+        except Exception:
+            evaluation = None
+    else:
+        evaluation = None
+
+    setattr(statement, 'wps_op_eval', evaluation)  # noqa: B010
```

Comment:
Let's move this side effect to `set_constant_evaluations`

Reply:
That makes sense, but I did it this way because it becomes a recursive functionality which expands and saves the evaluation to all the branches of a big operation, so that they don't have to be evaluated again. Moving this to `set_constant_evaluations` will mean that the evaluation will be reexecuted for all operations inside and will slow the execution

Reply:
Nevermind, I've already solved both these problems.